pwa-shell/src/ui/common/SwipeContentNavigator.js

```diff
@@ -0,0 +1,92 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { Gesture } from 'react-with-gesture';
+import { Redirect } from 'react-router-dom';
+
+class SwipeContentNavigator extends PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      hasSiblingContent: false,
+      xMin: 0,
+      xMax: 0,
+      openURL: '',
+    };
+    this.handleCursorMoving = this.handleCursorMoving.bind(this);
+  }
+
+  checkContentSiblings() {
+    const {
+      nextURL,
+      previousURL,
+    } = this.props;
+
+    let w = Math.max(document.documentElement.clientWidth, window.innerWidth || 0) / 2;
+    const hasNext = nextURL !== '';
+    const hasPrev = previousURL !== '';
+    this.setState({
```

Comment:
as a general rule of thumb, use state when it's really needed: in this case, instead of using `openURL` in state and then returing a `Redirect` component, I think it could be cleaner to use `history.push` (https://reacttraining.com/react-router/web/api/history)